src/client_conn.c

```diff
@@ -54,55 +56,50 @@ int get_socket() {
 
 static void tetris_translate(void *net_client, int x) {
 	char xdir = x > 0 ? 1 : 0;
-	char message[128];
-	sprintf(message, "%c%c", MSG_TYPE_TRANSLATE, xdir);
-	tetris_send_message((NetClient *)net_client, message);
+	char message[2];
+	sprintf(message, "%c", xdir);
+	tetris_send_message((NetClient *)net_client, message,
+	                    MSG_TYPE_TRANSLATE);
 }
 
 static void tetris_lower(void *net_client) {
-	char message[8];
-	sprintf(message, "%c", MSG_TYPE_LOWER);
-	tetris_send_message((NetClient *)net_client, message);
+	ttetris_net_request((NetClient *)net_client, NULL, 0, MSG_TYPE_LOWER);
 }
 
 static void tetris_rotate(void *net_client, int theta) {
 	char dir = theta > 0 ? 1 : 0;
-	char message[128];
-	sprintf(message, "%c%c", MSG_TYPE_ROTATE, dir);
-	tetris_send_message((NetClient *)net_client, message);
+	char message[2];
+	sprintf(message, "%c", dir);
+	tetris_send_message((NetClient *)net_client, message, MSG_TYPE_ROTATE);
 }
 
 static void tetris_drop(void *net_client) {
-	char message[128];
-	sprintf(message, "%c", MSG_TYPE_DROP);
-	tetris_send_message((NetClient *)net_client, message);
+	ttetris_net_request((NetClient *)net_client, NULL, 0, MSG_TYPE_DROP);
 }
 
 static void tetris_swap_hold(void *net_client) {
-	char message[128];
-	sprintf(message, "%c", MSG_TYPE_SWAP_HOLD);
-	tetris_send_message((NetClient *)net_client, message);
+	ttetris_net_request((NetClient *)net_client, NULL, 0,
+	                    MSG_TYPE_SWAP_HOLD);
 }
 
 StringArray *tetris_list(NetClient *net_client) {
-	char message[1];
-	message[0] = MSG_TYPE_LIST;
-	NetRequest *request = ttetris_net_request(net_client, message, 1);
+	NetRequest *request =
+	    ttetris_net_request(net_client, "", 0, MSG_TYPE_LIST);
```

Comment:
Curious. 2nd arg. NULL vs an empty string? Both seem to be used, is there a reason?

Reply:
Ahh; good catch. I'll update this to use NULL since an empty string is unnecessary bloat.